packages/web3-eth-contract/src/contract.ts

```diff
@@ -132,17 +172,132 @@ export class Contract<Abi extends ContractAbi>
 		this._jsonInterface = [...result] as unknown as Abi;
 	}
 
-	// eslint-disable-next-line class-methods-use-this
-	private _createContractFunction(
-		name: string,
-		// eslint-disable-next-line @typescript-eslint/no-unused-vars
-		_abi: AbiFunctionFragment,
-	): ContractBoundFunction {
-		return name;
+	private _createContractMethod(abi: AbiFunctionFragment): ContractBoundMethod {
+		return (...params) => {
+			validator.validate(abi.inputs, params);
+
+			if (abi.stateMutability === 'payable' || abi.stateMutability === 'pure') {
+				return {
+					arguments: params,
+					call: async (options?: PayableCallOptions, block?: BlockNumberOrTag) =>
+						this._contractMethodCall(abi, params, options, block),
+					// TODO: Use `web3-eth-tx` package to return `PromiEvent` instead.
+					send: async (options?: PayableCallOptions) =>
+						this._contractMethodSend(abi, params, options),
+					estimateGas: async (options?: PayableCallOptions, block?: BlockNumberOrTag) =>
+						this._contractMethodEstimateGas(abi, params, options, block),
+					encodeABI: () => encodeMethodABI(abi, params),
+				} as unknown as PayableMethodObject<unknown, unknown>;
+			}
+
+			return {
+				arguments: params,
+				call: async (options?: NonPayableCallOptions, block?: BlockNumberOrTag) =>
+					this._contractMethodCall(abi, params, options, block),
+				send: async (options?: NonPayableCallOptions) =>
+					this._contractMethodSend(abi, params, options),
+				estimateGas: async (options?: NonPayableCallOptions, block?: BlockNumberOrTag) =>
+					this._contractMethodEstimateGas(abi, params, options, block),
+				encodeABI: () => encodeMethodABI(abi, params),
+			} as unknown as NonPayableMethodObject<unknown, unknown>;
+		};
+	}
+
+	private async _contractMethodCall<Options extends PayableCallOptions | NonPayableCallOptions>(
+		abi: AbiFunctionFragment,
+		params: unknown[],
+		options?: Options,
+		block?: BlockNumberOrTag,
+	) {
+		return decodeMethodReturn(
+			abi,
+			await this.requestManager.send({
+				method: 'eth_call',
+				params: [
+					getTxCallParams({
+						abi,
+						params,
+						options,
+						contractOptions: this.options,
+					}),
+					block ?? BlockTags.LATEST,
+				],
+			}),
+		);
+	}
+
+	private async _contractMethodSend<Options extends PayableCallOptions | NonPayableCallOptions>(
+		abi: AbiFunctionFragment,
+		params: unknown[],
+		options?: Options,
+	) {
+		return decodeMethodReturn(
+			abi,
+			await this.requestManager.send({
+				method: 'eth_sendTransaction',
+				params: [
+					getSendTxParams({
+						abi,
+						params,
+						options,
+						contractOptions: this.options,
+					}),
+				],
+			}),
+		);
+	}
```

Comment:
https://github.com/ChainSafe/web3.js/issues/4685

Reply:
It's related to this message as well. https://github.com/ChainSafe/web3.js/pull/4678#discussion_r782113877